sw/device/tests/dif/dif_otbn_sanitytest.c

```diff
@@ -0,0 +1,252 @@
+extern const char _otbn_app_barrett384_dmem_start[];
+extern const char _otbn_app_barrett384_dmem_end[];
+extern const char _otbn_app_barrett384_wrap_barrett384[];
+
+const otbn_app_t kOtbnAppBarrett384 = {
+    .imem_start = _otbn_app_barrett384_imem_start,
+    .imem_end = _otbn_app_barrett384_imem_end,
+    .dmem_start = _otbn_app_barrett384_dmem_start,
+    .dmem_end = _otbn_app_barrett384_dmem_end,
+};
+
+const otbn_func_t kOtbnFuncBarrett384WrapBarrett384 = {
+    .app = &kOtbnAppBarrett384,
+    .func = _otbn_app_barrett384_wrap_barrett384,
+};
+
+const test_config_t kTestConfig = {
+    .can_clobber_uart = false,
+};
+
+static dif_otbn_t otbn;
+
+/*
+// Dump OTBN's memory. A development helper, unused otherwise.
+static void otbn_dump_dmem() {
+  uint32_t data[8];
+  for (int i = 0; i < dif_otbn_get_dmem_size_bytes(&otbn) / 32; ++i) {
+    dif_otbn_dmem_read(&otbn, i * 32, data, 32);
+    LOG_INFO("DMEM @%04d: 0x%08x%08x%08x%08x%08x%08x%08x%08x", i, data[7],
+             data[6], data[5], data[4], data[3], data[2], data[1], data[0]);
+  }
+}
+*/
+
+/**
+ * Load an application into OTBN
+ *
+ * Load the text and data segments into the instruction and data memories,
+ * respectively.
+ *
+ * @param app OTBN application to load
+ */
+static void load_app(const otbn_app_t *app) {
+  CHECK(app->imem_end >= app->imem_start);
+  const size_t imem_size = app->imem_end - app->imem_start;
+
+  CHECK(app->dmem_end >= app->dmem_start);
+  const size_t dmem_size = app->dmem_end - app->dmem_start;
+
+  LOG_INFO(
+      "Loading OTBN instruction memory image stored between address 0x%x and "
+      "%p (%d bytes)",
+      app->imem_start, app->imem_end, imem_size);
+  CHECK(imem_size > 0);
+  CHECK(imem_size % 4 == 0);
+  CHECK(dif_otbn_imem_write(&otbn, 0, (uint32_t *)app->imem_start, imem_size) ==
+            kDifOtbnOk,
+        "Unable to write IMEM application image (.text) to OTBN.");
+
+  if (dmem_size > 0) {
+    LOG_INFO(
+        "Loading OTBN data memory image stored between address %p and %p (%d "
+        "bytes)",
+        app->dmem_start, app->dmem_end, dmem_size);
+    CHECK(dif_otbn_dmem_write(&otbn, 0, (uint32_t *)app->dmem_start,
+                              dmem_size) == kDifOtbnOk,
+          "Unable to write DMEM application image (.data) to OTBN.");
+  } else {
+    LOG_INFO("No OTBN data memory image to load.");
+  }
+}
+
+/**
+ * Call a function on OTBN
+ *
+ * Set the entry point (start address) of OTBN to the desired function, and
+ * starts the OTBN operation.
+ *
+ * @param func the function to be called
+ */
+static void call_function(const otbn_func_t *func) {
+  uint32_t start_address = func->func - func->app->imem_start;
+  LOG_INFO("Calling function at address 0x%x on OTBN.", start_address);
+  CHECK(dif_otbn_start(&otbn, start_address) == kDifOtbnOk);
+}
+
+/**
+ * Busy wait for OTBN to be done with its operation.
+ */
+static void otbn_wait_for_done() {
+  bool busy = true;
+
```

Comment:
Nit: I'd drop this newline, since the busy def and the while loop should be part of the same stanza (same operation, same stanza).